oxenmq/message.h

```diff
@@ -51,7 +51,47 @@ class Message {
     /// Sends a request back to whomever sent this message.  This is effectively a wrapper around
     /// lmq.request() that takes care of setting up the recipient arguments.
     template <typename ReplyCallback, typename... Args>
-    void send_request(std::string_view cmd, ReplyCallback&& callback, Args&&... args);
+    void send_request(std::string_view command, ReplyCallback&& callback, Args&&... args);
+
+    /** Class returned by `send_later()` that can be used to call `back()`, `reply()`, or
+     * `request()` beyond the lifetime of the Message instance as if calling `msg.send_back()`,
+     * `msg.send_reply()`, or `msg.send_request()`.  For example:
+     *
+     *     auto send = msg.send_later();
+     *     // ... later, perhaps in a lambda or scheduled job:
+     *     send.reply("content");
+     *
+     * is equivalent to
+     *
+     *     msg.send_reply("content");
+     *
+     * except that it is valid even if `msg` is no longer valid.
+     */
+    class DeferredSend {
+    public:
+        OxenMQ& oxenmq; ///< The owning OxenMQ object
+        ConnectionID conn; ///< The connection info for routing a reply; also contains the pubkey/sn status
```

Comment:
should `conn` and `reply_tag` be const here?

Reply:
Fixed.

Reply:
Unfixed; leaving this non-const to allow moving it (`conn`, in particular, can have some allocated strings like the SN pubkey).